<%- include('./partials/header') -%> 

<form enctype="multipart/form-data">
    <h2>Log In</h2>
    <label for="email">Email</label>
    <input type="email" name="email" required>
    <div class="email error"></div>

    <label for="password">Password</label>
    <input type="password" name="password" required>
    <div class="password error"></div>

    <button>Log In</button>

    <header>
        <h3>Log in your account using</h3>
    </header>

    <main>
        <button>
            <a href="/auth/google" class="google-btn">Sign in with Google+ </a>
        </button>
        <br>
        <button>
            <a href="/auth/github" class="github-btn">Sign in with Github </a>
        </button>
        <br>
        <button>
            <a href="/auth/facebook" class="facebook-btn">Sign in with Facebook</a>
        </button>
        <br>
        <button>
            <a href="/auth/linkedin" class="linkedin-btn">Sign in with LinkedIn</a>
        </button>
    </main>
</form>

<%- include('./partials/footer') -%>

<script>
    const form = document.querySelector('form');
    const emailError = document.querySelector('.email.error');
    const passwordError = document.querySelector('.password.error');

    form.addEventListener('submit', async (e) => {
        e.preventDefault();

        emailError.textContent = '';
        passwordError.textContent = '';

        const email = form.email.value;
        const password = form.password.value;

        try {
            const res = await fetch('/login', {
                method: 'POST',
                body: JSON.stringify({ email, password }),
                headers: {
                    'Content-type': 'application/json',
                },
            });

            const data = await res.json();
            if(data.error){
                emailError.textContent = data.error.email;
                passwordError.textContent = data.error.password;
            }
            
            if(data.user){
                location.assign('/');
            }

        } catch (err) {
            console.log(err);
        }
    });
</script>